packages/backend/src/ee/models/AiAgentModel.ts

```diff
@@ -148,4 +152,143 @@ export class AiAgentModel {
+            organizationUuid: string;
+        },
+    ): Promise<AiAgent> {
+        return this.database.transaction(async (trx) => {
+            const [agent] = await trx(AiAgentTableName)
+                .insert({
+                    name: args.name,
+                    project_uuid: args.projectUuid,
+                    organization_uuid: args.organizationUuid,
+                    tags: args.tags,
+                    description: null,
+                    image_url: null,
+                })
+                .returning('*');
+
+            const integrationPromises =
+                args.integrations?.map(async (integration) => {
+                    switch (integration.type) {
+                        case 'slack':
+                            const [baseIntegration] = await trx(
+                                AiAgentIntegrationTableName,
+                            )
+                                .insert({
+                                    ai_agent_uuid: agent.ai_agent_uuid,
+                                    integration_type: integration.type,
+                                })
+                                .returning('*');
+
+                            await trx(AiAgentSlackIntegrationTableName).insert({
+                                ai_agent_integration_uuid:
+                                    baseIntegration.ai_agent_integration_uuid,
+                                organization_uuid: agent.organization_uuid,
+                                slack_channel_id: integration.channelId,
+                            });
+
+                            return {
+                                type: integration.type,
+                                channelId: integration.channelId,
+                            };
+                        default:
+                            return assertUnreachable(
+                                integration.type,
+                                `Unknown integration type ${integration.type} in createAgent`,
+                            );
+                    }
+                }) || [];
+
+            // Wait for all integration creation operations to complete
+            const integrations = await Promise.all(integrationPromises);
+
+            return {
+                uuid: agent.ai_agent_uuid,
+                name: agent.name,
+                projectUuid: agent.project_uuid,
+                organizationUuid: agent.organization_uuid,
+                tags: agent.tags,
+                integrations,
+            };
+        });
+    }
+
+    async updateAgent(
+        args: Pick<
+            ApiUpdateAiAgent,
+            'name' | 'projectUuid' | 'tags' | 'integrations'
+        > & {
+            agentUuid: string;
+            organizationUuid: string;
+        },
+    ): Promise<AiAgent> {
+        return this.database.transaction(async (trx) => {
+            const [agent] = await trx(AiAgentTableName)
+                .where('ai_agent_uuid', args.agentUuid)
```

Comment:
```suggestion
                .where({
                   'ai_agent_uuid': args.agentUuid,
                   'organization_uuid': args.organizationUuid
                })
```
this way we combine the two filters directly!

Reply:
moving the org uuid check to service now